### ISSUE TITLE
Get-ChildItem doesn't treat directory symlinks as directories on Unix

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#3723]
Note: Applies to Linux and macOS, not Windows.

In most contexts, a symlink to a _directory_  should be treated as a directory.

`Get-ChildItem` on Unix platform currently treats directory symlinks as _files_, and therefore doesn't show their _contents_.

Steps to reproduce (run on a Unix platform)
------------------

```powershell
# Create symlink 'tmpl' to directory '/tmp' in the current dir.
ln -s /tmp tmpl
# Pass the directory symlink to Get-ChildItem
Get-ChildItem tmpl 
```

Expected behavior
-----------------

A listing of the _contents_ of directory `/tmp`

Actual behavior
---------------

Something like the following is output; i.e., information about the _symlink itself_, not its target's contents:

```
    Directory: /Users/jdoe


Mode                LastWriteTime         Length Name                                                                                                                              
----                -------------         ------ ----                                                                                                                              
d----l           5/7/17   2:35 PM                tmpl                                                                                                                              
```

Environment data
----------------

```powershell
PowerShell Core v6.0.0-alpha (v6.0.0-alpha.18) on macOS 10.12.4
```

**Comment by iSazonov** [comment]
It is duplicate #1958 and fixed by @jeffbi in #3697

**Comment by mklement0** [comment]
@iSazonov  Glad to hear it; thanks for letting me know.
